docs/api/process.md

```diff
@@ -134,6 +135,28 @@ Each frame has its own JavaScript context. When contextIsolation is enabled, the
 world also has a separate JavaScript context.
 This property is only available in the renderer process.
 
+### `process.parentPort`
+
+A `MessagePort` if this is a [`UtilityProcess`](utility-process.md) allowing communication
```

Comment:
```suggestion
A [`MessagePortMain`](message-port-main.md) if this is a [`UtilityProcess`](utility-process.md) (or `null` otherwise), allowing communication
```

Reply:
What happens if this port gets transferred to another process? Should that be possible?

Reply:
`parentPort` is not meant to be transferrable to another process, the current implementation exposes `parentPort` as an event emitter with certain properties of `MessagePort` added to it that allows communication with parent process. If the object gets transferred to another process I expect the `postMessage` calls to no-op. It also does not support `start` and `close` methods of `MessagePort`, I wonder if the `MessagePort` type is even a correct declaration. I should align the methods and events exposed on this object to match `MessagePortMain`.

```
interface ParentPort extends EventEmitter {
    postMessage(message: any): void;
    on(eventName: 'message', listener: (event: any, ports: any[]) => void): this;
}
```

Reply:
Yeah, seems like an incorrect type then. I'm fine with it not being a "real" messageport (& thus non transferrable). We should probably have a mechanism for queuing messages until the process installs a listener though, see https://github.com/electron/electron/pull/34980#discussion_r980348345. MessagePort / MessagePortMain does this automatically under the hood, so we'll need to replicate that behavior.